### PR TITLE
Added ellipse

### PR DESCRIPTION
**Pull request by ROCKFAL1**
Added ellipse to library and it's example to emulator

### CODE REVIEW

File: emulator/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#Folder
+target
```
Comment (Gordon01) [inline emulator/.gitignore]: This shouldn't be needed, but even if it is, the root .gitignore should be edited.

File: rugui/src/geometry.rs [hunk 2]
```diff
@@ -188,3 +199,92 @@ impl Circle {
         }
     }
 }
+
+impl Ellipse {
+    pub fn new(width: u32, height: u32, center: Vec2, color: Color) -> Self {
```
Comment (Gordon01) [inline rugui/src/geometry.rs]: Can we merge the code from Circle and Ellipse and reuse it as much as we can?
Currently, `Circle::new()`, `Circle::from_bbox()` and `Circle::filled()` are almost identical to the Ellipse ones.

File: rugui/src/geometry.rs [hunk 3]
```diff
@@ -188,3 +199,92 @@ impl Circle {
...
+                        canvas.draw_pixel(x + dx, y + dy, &self.color);
+                    }
+                }              
+                
+            }
+        }
+
+    }
+
+}
```
Comment (Gordon01) [inline rugui/src/geometry.rs]: Most codebases add empty lines at the end of every file.

File: rugui/src/geometry.rs [hunk 4]
```diff
@@ -188,3 +199,92 @@ impl Circle {
...
+            self.thickness = self.max_thickness();
+        } else {
+            self.thickness = 1;
+        }
+
+        self
+    }
+
+    pub fn max_thickness(&self) -> u32 {
+        if self.height >= self.width {
```
Comment (Gordon01) [inline rugui/src/geometry.rs]: `self.height.min(self.width)`
should do the job

File: rugui/src/geometry.rs [hunk 5]
```diff
@@ -188,3 +199,92 @@ impl Circle {
...
+            self.width
+        }
+        else {
+            self.height
+        }
+    }
+
+    pub fn thickness(mut self, t: u32) -> Self {
+        self.thickness = t;
```
Comment (Gordon01) [inline rugui/src/geometry.rs]: `self.thickness = t.min(self.max_thickness());`
or something like that

**Review by Gordon01 — CHANGES_REQUESTED** [review]
Thank you for PR, but it needs some polish!

File: rugui/src/geometry.rs (hunk 5 above)
Comment (ROCKFAL1) [reply rugui/src/geometry.rs]: Fixed

**Pull request merged**